### PR TITLE
json issue and the array-set helper

### PR DESCRIPTION
**Pull request by kkamkou**
Hello Jonathan,
Another small part of code. Please, review.

### CODE REVIEW

**Comment by a-musing-moose** [comment]
Hi Kkamkou,

The setFromArray() function I think is unneeded.  It does pretty much the same thing as __setData() or am I missing something?

**Comment by kkamkou** [comment]
Hello Jonathan,
__setData creates new Generic property if it doesn't exist. setFromArray is just useful helper (for me, but you should decide).

File: src/morph/Iterator.php [hunk 1]
```diff
@@ -80,23 +85,22 @@ public function toCollection()
...
+     * @param  array $item
+     * @return Object
      */
     private function createObject(array $item)
     {
-        $class = \get_class($this->type);
+        $class = get_class($this->type);
         $object = new $class;
-        $object->__setData($item, Enum::STATE_CLEAN)
-            ->collection($this->type->collection());
```
Comment (kkamkou) [inline src/morph/Iterator.php:91]: it should be here.

**Comment by a-musing-moose** [comment]
kkamkou - It seems you have committed other code to your repo since opening this pull request. As a result this pull request also includes those commits which shouldn't be merged in.

Also could we not add an optional flag '$addNewProperties = true' to __setData rather than add a new function? If the default is true it would be backward compatible and in your case you just set it to false?

So if you would like this feature please can you add it as an optional feature of __setData in a fresh branch by itself and open another pull request?

**Pull request closed without merging**